crates/shared/src/sources/balancer_v2/pools/common.rs

```diff
@@ -77,6 +94,59 @@ impl PoolInfoFetching for PoolInfoFetcher {
             block_created,
         })
     }
+
+    fn fetch_common_pool_state(
+        &self,
+        pool: &PoolInfo,
+        batch: &mut Web3CallBatch,
+        block: BlockId,
+    ) -> BoxFuture<'static, Result<PoolState>> {
+        let pool_contract = self.base_pool_at(pool.address);
+        let paused = pool_contract
+            .get_paused_state()
+            .block(block)
+            .batch_call(batch);
+        let swap_fee = pool_contract
+            .get_swap_fee_percentage()
+            .block(block)
+            .batch_call(batch);
+        let balances = self
+            .vault
+            .get_pool_tokens(Bytes(pool.id.0))
+            .block(block)
+            .batch_call(batch);
+
+        // Because of a `mockall` limitation, we **need** the future returned
+        // here to be `'static`. This requires us to clone and move `pool` into
+        // the async closure - otherwise it would only live for as long as
+        // `pool`, i.e. `'_`.
+        let pool = pool.clone();
+        async move {
+            let (paused, _, _) = paused.await?;
+            let swap_fee = Bfp::from_wei(swap_fee.await?);
+
+            let (token_addresses, balances, _) = balances.await?;
+            ensure!(pool.tokens == token_addresses, "pool token mismatch");
```

Comment:
Was this constraint introduced here? Do you have reason to believe this would/could ever happen?

Reply:
It was introduced here. It was just more of an "this is an in-code documentation of an assumption we make" kind of thing.